pkg/util/version/stream.go

```diff
@@ -14,33 +14,36 @@ type Stream struct {
 }
 
 // GetUpgradeStream determines if a valid upgrade path is available, and if so, returns the corresponding stream.
-func GetUpgradeStream(v *Version) (*Stream, error) {
-	// ARO version matches OCP version - X.Y.Z.
+func GetUpgradeStream(v *Version, upgradeY bool) (*Stream, error) {
+	// ARO version matches OCP version - X.Y.Z
 	// We know we can have only single version configured per Y release. These
-	// version should be edge points for Y+1 upgrades.
+	// versions should be edge points for Y+1 upgrades.
 
-	// We check first with which configured Y stream we are dealing with
+	// We first check which configured Y stream we are dealing with
 	for _, upgradeCandidate := range Streams {
-		if upgradeCandidate.Version.V[0] == v.V[0] &&
-			upgradeCandidate.Version.V[1] == v.V[1] {
+		if upgradeCandidate.Version.V[0] == v.V[0] && // Major (X) version is the same
+			upgradeCandidate.Version.V[1] == v.V[1] { // Minor (Y) version is the same
 
-			// we DO NOT upgrade if CVO version is already higher
+			// We DO NOT upgrade if CVO (cluster version operator) version is already higher
 			if upgradeCandidate.Version != nil && upgradeCandidate.Version.Lt(v) {
-				return nil, fmt.Errorf("not upgrading: cvo desired version is %s", v)
+				return nil, fmt.Errorf("not upgrading: CVO desired version is %s", v)
 			}
 
-			// If upgradeCandidate is higher than CVO - use it to upgrade to ARO
-			// latest x.Y release before jumping major version.
+			// If upgradeCandidate is higher than CVO, use it to upgrade (to latest Z version)
 			if v.Lt(upgradeCandidate.Version) {
 				return &upgradeCandidate, nil
 			}
 
-			// If we on right version, we need to upgrade next major version
+			// If the versions match, we need to upgrade minor version,
+			// provided that we are allowed to (i.e. upgradeY is true)
 			if v.Eq(upgradeCandidate.Version) {
 				for _, upgradeCandidate := range Streams {
-					// if incoming version is 4.2, we return 4.3 for major upgrade.
+					// e.g. if incoming version is 4.2, we return 4.3 for Y version upgrade.
 					if upgradeCandidate.Version.V[1] == v.V[1]+1 {
-						return &upgradeCandidate, nil
+						if upgradeY {
```

Comment:
the following example behaviour is currently implemented, but is it actually wanted?
rp baseline = 4.4.27
cluster = 4.4.28
upgradeY = true
"not upgrading" error returned rather than upgrading the cluster to 4.5.x

Reply:
@jim-minter I don't think so -- I will check if `upgradeY` is false before returning the error above.